core/templates/dev/head/galleries/Gallery.js

```diff
@@ -201,6 +204,115 @@ oppia.controller('Gallery', [
+      $scope.galleryTilesResizedWidth = $scope.tileDisplayCount *
+        TILE_DISPLAY_WIDTH;
+      if ($scope.galleryTilesResizedWidth >
+          TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY) {
+        $scope.galleryTilesResizedWidth = TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY;
+        $scope.tileDisplayCount = MAX_TILE_DISPLAY;
+      }
+      $('.oppia-gallery-tiles-carousel').css({
+        width: $scope.galleryTilesResizedWidth + 'px'
+      });
+
+      // The following determines whether to enable left scroll after resize.
+      for (i = 0; i < $('.oppia-gallery-tiles-carousel').length; i++) {
+        var carouselJQuerySelector = '.oppia-gallery-tiles:eq(n)'.replace(
+          'n', i);
+        var carouselScrollPositionPx = $(carouselJQuerySelector).scrollLeft();
+        var index = Math.ceil(carouselScrollPositionPx / TILE_DISPLAY_WIDTH);
+        $scope.rowIndices[i] = index;
+      }
+    };
+
+    $scope.scroll = function(ind, isLeftScroll) {
+      var carouselJQuerySelector = '.oppia-gallery-tiles:eq(n)'.replace(
+        'n', ind);
+      var lftOverlaySelector =
+        '.oppia-gallery-tiles-carousel-overlay-left:eq(n)'.replace('n', ind);
+      var rtOverlaySelector =
+        '.oppia-gallery-tiles-carousel-overlay-right:eq(n)'.replace('n', ind);
+
+      var direction = isLeftScroll ? -1 : 1;
+      var carouselScrollPositionPx = $(carouselJQuerySelector).scrollLeft();
+
+      // Scrolling eligibility checks.
+      if ($scope.galleryGroups[ind].activity_summary_dicts.length <=
+          $scope.tileDisplayCount) {
+        return;
+      } else if (isLeftScroll && $scope.rowIndices[ind] == 0) {
+        return;
+      } else if (!isLeftScroll && $scope.rowIndices[ind] >=
+                 $scope.galleryGroups[ind].activity_summary_dicts.length -
+                 $scope.tileDisplayCount) {
+        return;
+      }
+
+      carouselScrollPositionPx = Math.max(0, carouselScrollPositionPx);
+
+      if (isLeftScroll) {
+        $scope.rowIndices[ind] = Math.max(0,
+              $scope.rowIndices[ind] - $scope.tileDisplayCount);
+      } else {
+        $scope.rowIndices[ind] = Math.min(
+          $scope.galleryGroups[ind].activity_summary_dicts.length -
+          $scope.tileDisplayCount + 1,
+          $scope.rowIndices[ind] + $scope.tileDisplayCount);
+      }
+      var newScrollPositionPx = carouselScrollPositionPx +
+        ($scope.galleryTilesResizedWidth * direction);
+
+      $(carouselJQuerySelector).animate({
+        scrollLeft: newScrollPositionPx
+      }, {
+        duration: 800,
+        queue: false
+      });
+
+      $(lftOverlaySelector).css({
+        display: 'inline'
+      }).fadeOut({
+        duration: 800,
+        queue: false
+      });
+      $(rtOverlaySelector).css({
+        display: 'inline'
+      }).fadeOut({
+        duration: 800,
+        queue: false
+      });
+    };
+
+    $(window).resize(function() {
+      initCarousels();
```

Comment:
Try adding $scope.$apply() after this. Does it fix the slow-update issue? (I narrowed it down to it being $scope.tileDisplayCount that was updating too slowly -- it's not due to ng-hide: I wrote <[tileDisplayCount]> in the HTML and that updated slowly too.)